modules/host/contractmanager/writeaheadlogsync.go

```diff
@@ -0,0 +1,281 @@
+package contractmanager
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/NebulousLabs/Sia/persist"
+)
+
+// syncResources will call Sync on all resources that the WAL has open, except
+// for the WAL temp file, which needs to be sync'd separately.
+func (wal *writeAheadLog) syncResources() {
+	// The syncing is done in parallel to ensure that the total amount of time
+	// consumed here is approximately equivalent to the cost of Syncing the
+	// slowest file, instead of waiting to sync some files until other files
+	// have finished syncing.
+	var wg sync.WaitGroup
+	// Queue a thread to sync + copy-on-write the settings file.
+	wg.Add(1)
+	go func() {
+		tmpFilename := filepath.Join(wal.cm.persistDir, settingsFileTmp)
+		filename := filepath.Join(wal.cm.persistDir, settingsFile)
+		// Sync the settings file.
+		err := wal.fileSettingsTmp.Sync()
+		if err != nil {
+			wal.cm.log.Critical("ERROR: unable to sync the contract manager settings:", err)
+			panic("unable to sync contract manager settings, crashing to avoid data corruption")
+		}
+		err = wal.fileSettingsTmp.Close()
+		if err != nil {
+			wal.cm.log.Println("unable to close the temporary contract manager settings file:", err)
+		}
+		// COW the settings file.
+		err = os.Rename(tmpFilename, filename)
```

Comment:
oh. I have misused the term copy-on-write all over the documentation. Guess I have to clean that up.